### PR TITLE
#2471 Include in download directory tflint config

### LINKED ISSUE #2471: Could not find .tflint.hcl config file in the parent folders when using TERRAGRUNT_DOWNLOAD env
Reproducer:
```
mkdir demo
mkdir -p modules/foo
touch modules/foo/main.tf
cat > demo/.tflint.hcl <<-END
  config { format = "compact" }
END
cat > demo/terragrunt.hcl <<-END
  terraform {
    source = "../modules//foo"
    before_hook "lint" {
      commands = ["validate"]
      execute  = ["tflint"]
    }
  }
END
TERRAGRUNT_DOWNLOAD=/tmp/.terragrunt-cache terragrunt validate --terragrunt-working-dir demo
```

### PR DESCRIPTION
**Pull request by denis256**
## Description

Update terragrunt to include in download directory tflint configuration file.

Fixes #2471 .

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [x] Update the docs.
- [ ] Run the relevant tests successfully, including pre-commit checks.
- [x] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [x] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes (draft)

Updated terragrunt to copy in download directory tflint configuration files.

### CODE REVIEW

File: test/integration_tflint_test.go [hunk 1]
```diff
@@ -25,7 +27,9 @@ func TestTflintFindsNoIssuesWithValidCode(t *testing.T) {
 	assert.NotContains(t, errOut.String(), "Error while running tflint with args:")
 	assert.NotContains(t, errOut.String(), "Tflint found issues in the project. Check for the tflint logs above.")
 
-	assert.Contains(t, errOut.String(), fmt.Sprintf("--config %s/.tflint.hcl", modulePath))
+	found, err := regexp.MatchString(fmt.Sprintf("--config %s/.*/.tflint.hcl", modulePath), errOut.String())
```
Comment (marinalimeira) [inline test/integration_tflint_test.go:30]: So this test wasn't testing?

**Review by marinalimeira — APPROVED** [review]
The code changes LGTM. I haven't tested the changes, should we ask the issue reporter to test it?

**Comment by Etiene** [comment]
Quick question, is this a file going in the cached copies of the modules? And does it go on every subdir? Or only the root dir of the copy?

File: test/integration_tflint_test.go (hunk 1 above)
Comment (denis256) [reply test/integration_tflint_test.go]: Before my changes, `.tflint.hcl` wasn't included in `.terragrunt-cache` directory and was loaded from "modulePath", with my changes it is copied to `.terragrunt-cache` but module path is hashed so I added regexp to match path

**Comment by denis256** [comment]
Copy is performed in same directory where it is stored:

```
.
├── .terraform.lock.hcl
├── .terragrunt-cache
│   └── 7-krBkt1wcuSxg_MU7cDkTJCt4A
│       └── FKIM0dVsZSB7ZaKQlJdcfEHSQHI
│           ├── .terraform
│           │   └── providers
│           │       └── registry.terraform.io
│           │           └── hashicorp
│           │               └── random
│           │                   └── 3.4.3
│           │                       └── linux_amd64
│           │                           └── terraform-provider-random_v3.4.3_x5
│           ├── .terraform.lock.hcl
│           ├── .terragrunt-module-manifest
│           ├── .terragrunt-source-manifest
│           ├── .terragrunt-source-version
│           ├── .tflint.hcl <-----------------------------------
│           ├── d1
│           │   ├── .terragrunt-module-manifest
│           │   ├── .terragrunt-source-manifest
│           │   └── file.txt
│           ├── main.tf
│           ├── outputs.tf
│           ├── terragrunt.hcl
│           └── variables.tf
├── .terragrunt-test
├── .tflint.hcl <---------------------------------------------
├── d1
│   ├── .terragrunt-test
│   └── file.txt
├── main.tf
├── outputs.tf
├── terragrunt.hcl
└── variables.tf
```

**Review by brikis98 — APPROVED** [review]
LGTM!

**Pull request merged**